holoviews/plotting/bokeh/chart.py

```diff
@@ -863,3 +864,203 @@ def get_batched_data(self, element, ranges, empty):
                             vdims=el.vdims)
         return self.get_data(collapsed, ranges, empty)
+
+
+
+class BoxWhiskerPlot(CompositeElementPlot, ColorbarPlot, LegendPlot):
+
+    color_index = param.ClassSelector(default=None, class_=(basestring, int),
+                                      allow_None=True, doc="""
+      Index of the dimension from which the color will the drawn""")
+
+    show_legend = param.Boolean(default=False, doc="""
+        Whether to show legend for the plot.""")
+
+    # X-axis is categorical
+    _x_range_type = FactorRange
+
+    # Declare that y-range should auto-range if not bounded
+    _y_range_type = DataRange1d
+
+    # Map each glyph to a style group
+    _glyph_styles = {'rect': 'whisker', 'segment': 'whisker',
+                     'vbar': 'box', 'circle': 'outlier', 'hbar': 'box'}
+
+    # Define all the glyphs making up the plot
+    _glyphs = ['vbar_1', 'vbar_2', 'segment_1', 'segment_2',
+               'rect_1', 'rect_2', 'circle', 'hbar_1', 'hbar_2']
+
+    # Define all the glyph handles to update
+    _update_handles = (lambda gs:
+                       [glyph+'_'+model for model in ['glyph', 'glyph_renderer', 'source']
+                        for glyph in gs] + ['color_mapper', 'colorbar'])(_glyphs)
+
+    style_opts = (['whisker_'+p for p in line_properties] +\
+                  ['box_'+p for p in fill_properties+line_properties] +\
+                  ['outlier_'+p for p in fill_properties+line_properties] + ['width', 'cmap'])
+
+    def get_extents(self, element, ranges):
+        """
+        Extents are set to '' and None because x-axis is categorical and
+        y-axis auto-ranges.
+        """
+        return ('', None, '', None)
+
+    def _get_axis_labels(self, *args, **kwargs):
+        """
+        Override axis labels to group all key dimensions together.
+        """
+        element = self.current_frame
+        xlabel = ', '.join([kd.pprint_label for kd in element.kdims])
+        ylabel = element.vdims[0].pprint_label
+        return xlabel, ylabel, None
+
+    def _glyph_properties(self, plot, element, source, ranges):
+        properties = dict(self.style[self.cyclic_index], source=source)
+        if self.show_legend and not element.kdims:
+            properties['legend'] = element.label
+        return properties
+
+    def _get_factors(self, element):
+        """
+        Get factors for categorical axes.
+        """
+        if not element.kdims:
+            return [element.label], []
+        else:
+            factors = [', '.join([d.pprint_value(v).replace(':', ';')
+                                  for d, v in zip(element.kdims, key)])
+                       for key in element.groupby(element.kdims).data.keys()]
+            if self.invert_axes:
+                return [], factors
+            else:
+                return factors, []
+
+    def get_data(self, element, ranges=None, empty=False):
```

Comment:
This method is really long. If it can be broken up into sensible chunks, then that might help readability.